specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,1026 @@
+developer adds an attribute to the image element and sets the text fields in the payload.
+
+```cpp
+// Uses an API template to Compose and Display a toast
+void CreateToastPayload()
+{
+    auto doc = ToastNotificationManager::Default().GetXmlTemplateContent(ToastTemplateType::ToastWithAvatar);
+    auto srcAttribute = doc.CreateAttribute(L"src");
+    srcAttribute.Value(L"images/toastImageAndText.png");
+    auto images = doc.GetElementsByTagName(L"image");
+    images.GetAt(0).Attributes().SetNamedItem(srcAttribute);
+
+    auto textNodes = doc.GetElementsByTagName(L"text");
+
+    for (auto&& textNode : textNodes)
+    {
+        textNode.AppendChild(doc.CreateTextNode(L"This is some notification text!"));
+    }
+
+    DisplayToast(doc);
+}
+```
+
+## Processing a Toast Activation
+
+The app will need to process the activator in response to a User activating the toast. 2 common
+scenarios here are
+
+1. Have the app launch in the foreground in a specific UI context OR
+2. Have the app process a toast action (like a button press in the toast body) in the background.
+
+```cpp
+//<toast launch="action=openThread&amp;threadId=92187">
+//  <visual>
+//      <binding template="ToastGeneric">
+//          <text hint-maxLines="1">Jill Bender</text>
+//          <text>Check out where we camped last weekend! Incredible!!</text>
+//          <image placement="hero" src="https://unsplash.it/360/180?image=1043" />
+//      </binding>
+//  </visual>
+//  <actions>
+//      <input id="textBox" type="text" placeHolderContent="reply" / >
+//      <action
+//          content="Send"
+//          imageUri="Assets/Icons/send.png"
+//          hint-inputId="textBox"
+//          activationType="background"
+//          arguments="action=reply&amp;threadId=92187" />
+//  </actions>
+//</toast>
+
+const winrt::hstring c_toastLaunchAction = L"action=openThread&amp;threadId=92187";
+const winrt::hstring c_toastReplyButtonAction = L"action=reply&amp;threadId=92187";
+
+void ProcessToastArgs(ToastActivatedEventArgs const& toastActivatedEventArgs)
+{
+    if (toastActivatedEventArgs.ActivationArgs() == c_toastLaunchAction)
+    {
+        // The user clicks on the toast: So use the launchAction to do stuff
+        // Do LaunchAction Stuff
+    }
+    else if (toastActivatedEventArgs.ActivationArgs() == c_toastReplyButtonAction)
+    {
+        // The user clicked on the reply button on the toast. So query the input field
+        auto input = toastActivatedEventArgs.UserInput();
+        auto replyBoxText = input.Lookup(L"ReplyBox");
+
+        // Process the reply text
+        ProcessReply(replyBoxText);
+    }
+}
+```
+
+## Processing Toast History
+
+The ToastHistory component is always used to perform Delete and Get operations on behalf of the app.
+
+```cpp
+// Get a List of all toasts from the Action Centre
+winrt::Windows::Foundation::IAsyncOperation<Windows::Foundation::Collections::IVector<ToastNotification> > GetToastListASync()
```

Comment:
Modern compilers don't require separating the template closing `>` anymore, so you can write:
```suggestion
winrt::Windows::Foundation::IAsyncOperation<Windows::Foundation::Collections::IVector<ToastNotification>> GetToastListASync()
```